CommandLine/emake/Main.cpp

```diff
@@ -63,14 +69,23 @@ int main(int argc, char* argv[])
       std::string ext;
       size_t dot = input_file.find_last_of('.');
       if (dot != std::string::npos) ext = tolower(input_file.substr(dot + 1));
-      if (ext != "sog") {
+      if (ext == "sog") {
+        if (!ReadSOG(input_file, &game)) return 1;
+      } else if (ext == "gmx") {
+        boost::filesystem::path p = input_file;
```

Comment:
all this logic does is translate game.gmx to game.gmx/game.project.gmx so the cli can accept either. I think it's fine here

Reply:
Please use C++17: https://stackoverflow.com/a/43281413

Reply:
no. emake already uses boost for other shit and we dont support c++17. This would require upgrading gcc on every travis and would up our build time to ~two weeks

Reply:
Josh has directory functions that can be included from CompilerSource without adding boost into the CompilerSource. I'm also wanting to switch us to deepflags or something else available on pacman. I absolutely hate boost right now because it's not doing enough for us. Unless you decide to start using it for image and sounds too, then I'll reconsider. But for right now, we're simply not getting enough use of it to justify its inclusion.